File: template/template.cs
namespace aoc2024;

public class DayX : Day<int, int, int, int> {

    public (int P1, int P2) Prepare(string in_string) {
        return (0, 0);
    }

    public int Part1(int in_data) {
        return 0;
    }

    public int Part2(int in_data) {
        return 0;
    }

    public (object P1, object P2) Run() {
        var input_raw = File.ReadAllText("input/dayX.txt");
        var input_clean = Prepare(input_raw);
        return (Part1(input_clean.P1), Part2(input_clean.P2));
    }
}

File: template/template.kt
internal typealias InputX = Any
internal typealias OutputX = Any

class DayX: Day<InputX, InputX, OutputX, OutputX> {

	override fun prepare(inString: String): Pair<InputX, InputX> {
		return Pair(0, 0)
	}

	override fun part1(inData: InputX): OutputX {
		return 0
	}

	override fun part2(inData: InputX): OutputX {
		return 0
	}

	override fun run(): Pair<OutputX, OutputX> {
		val inputRaw = tool.readInput("dayX")
		val inputClean = prepare(inputRaw)
		return Pair(part1(inputClean.first), part2(inputClean.second))
	}
}